### ISSUE TITLE
dune forwards signal to process but sends SIGKILL right away

### ISSUE THREAD

**Issue by smondet** [ocaml/dune#2445]
If a test is running with `dune` and Ctrl-C is hit, the test receives the SIGINT but gets killed with SIGKILL a couple of milliseconds later (so the test's attempts to clean-up by catching SIGINT/SIGTERM are hopeless).

SIGKILL is there:
https://github.com/ocaml/dune/blob/823c74063ee6683d5f2c378a4fbbc3ef7c7a3086/src/scheduler.ml#L589-L593

Replacing `Sys.sigkill` with `Sys.sigterm` seems to solve the problem for me, but I guess the right solution involves sending `sigterm`, waiting for processes to die until a given `TIMEOUT`, and then kill them again with `sigkill`, right?

**Comment by diml** [comment]
> Replacing Sys.sigkill with Sys.sigterm seems to solve the problem for me, but I guess the right solution involves sending sigterm, waiting for processes to die until a given TIMEOUT, and then kill them again with sigkill, right?

That seems right, yes.

**Comment by Alizter** [comment]
@robinbb Did you intend to close this?

**Comment by robinbb** [comment]
> [@robinbb](https://github.com/robinbb) Did you intend to close this?

No. I had thought that my modified PR description would not result in this issue closing, but it did. :-(

**Comment by robinbb** [comment]
> > [@robinbb](https://github.com/robinbb) Did you intend to close this?
> 
> No. I had thought that my modified PR description would not result in this issue closing, but it did. :-(

To be clear, I was fixing in when I wrote that. :-) It is corrected, now: the issue is open, and there is no longer a faulty changelog checked in. Also, I am working on this issue in draft PR #14224 .
